pysindy/differentiation/sindy_derivative.py

```diff
@@ -76,9 +78,16 @@ def _differentiate(self, x, t=1):
         differentiator = methods[self.kwargs["kind"]](
             **{k: v for k, v in self.kwargs.items() if k != "kind"}
         )
-        x_dot = differentiator.d(x, t, axis=0)
+        x = np.moveaxis(x, self.axis, 0)
+        x_shape = x.shape
+        flat_x = x.reshape((t.size, int(x.size / t.size)))
+        flat_x_dot = differentiator.d(flat_x, t, axis=0)
```

Comment:
This is the line where `AxesArray` gets converted back to a `numpy.ndarray`, `flat_x` is an `AxesArray`, `flat_x_dot` is `numpy.ndarray`